### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by LDSamson**
Nothing special, this just tests some edge cases that were previously not yet covered by unit tests. Hardly any changes are made in the actual code base.

It also places the function db_update_version to the dev folder since we do not plan to support this function in the current form; it is only meant to update from the old database format that hopefully no one else but us used, and it is not needed for setting up a new study with the application.

### CODE REVIEW

File: tests/testthat/test-fct_SQLite.R [hunk 1]
```diff
@@ -386,6 +423,26 @@ describe(
...
+  it("returns an empty string if the version is not available", {
+    temp_path <- withr::local_tempfile(fileext = ".sqlite") 
+    con <- get_db_connection(temp_path)
+    expect_equal(db_get_version(temp_path), "")
+  })
+  it("errors if database is uavailable", {
+    expect_error(db_get_version("fake_path/db.sqlite"))
+  })
+})
+
```
Comment (aclark02-arcus) [inline tests/testthat/test-fct_SQLite.R:445]: Do we need tests for functions that aren't in `R/`?

File: R/fct_appdata.R [hunk 2]
```diff
@@ -284,14 +284,10 @@ get_appdata <-  function(
...
     item_group_x <- unique(x$item_group)
-    if(length(item_group_x) != 1) stop(
-      "item_group consists of multipe elements which is not allowed: ", 
-      item_group_x
-    )
     form_type_x <- unique(with(var_levels, form_type[item_group == item_group_x]))
     if(length(form_type_x) != 1) stop(
       "form_type consists of multipe elements which is not allowed: ", 
-      form_type_x
+      paste0(form_type_x, collapse = ", ")
```
Comment (aclark02-arcus) [inline R/fct_appdata.R:290]: ha, we already made this change on our side!

**Review by aclark02-arcus — APPROVED** [review]

File: tests/testthat/test-fct_SQLite.R (hunk 1 above)
Comment (LDSamson) [reply tests/testthat/test-fct_SQLite.R:445]: I think these functions are available in `R/`, right?

Comment (aclark02-arcus) [reply tests/testthat/test-fct_SQLite.R:445]: Oh sorry, I was confusing `get_db_version()` with `update_db_version()` which now lives in `dev/`
